crates/typst-library/src/foundations/str.rs

```diff
@@ -286,6 +288,31 @@ impl Str {
         Ok(c.into())
     }
 
+    /// Normalizes the string to the given Unicode Normalization Form. This is useful when
+    /// manipulating strings containing combining Unicode characters.
+    ///
+    /// ```example
+    /// #"é".normalize("NFD") \ // "e\u{0301}"
+    /// #"ſ́".normalize("NFKC") // "ś", = "\u{015b}"
+    /// ```
+    #[func]
+    pub fn normalize(
+        &self,
+        /// One of `"NFC"`, `"NFD"`, `"NFKC"`, or `"NFKD"`, specifiying the [Unicode Normalization
+        /// Form](https://unicode.org/reports/tr15/#Norm_Forms) to use. If set to `auto`, NFC is used.
+        form: Smart<UnicodeNormalForm>,
+    ) -> Str {
+        match form {
+            Smart::Auto => self.nfc().collect(),
+            Smart::Custom(nf) => match nf {
+                UnicodeNormalForm::Nfc => self.nfc().collect(),
+                UnicodeNormalForm::Nfd => self.nfd().collect(),
+                UnicodeNormalForm::Nfkc => self.nfkc().collect(),
+                UnicodeNormalForm::Nfkd => self.nfkd().collect(),
+            },
+        }
```

Comment:
Why not this?

```suggestion
        match form.unwrap_or(UnicodeNormalForm::Nfc) {
            UnicodeNormalForm::Nfc => self.nfc().collect(),
            UnicodeNormalForm::Nfd => self.nfd().collect(),
            UnicodeNormalForm::Nfkc => self.nfkc().collect(),
            UnicodeNormalForm::Nfkd => self.nfkd().collect(),
        }
```

Reply:
Ah didn't know Smarts could be unwrapped, thanks